protocol/op-stack-manager-arch.md

```diff
@@ -0,0 +1,292 @@
+3. The `protocolVersionsProxy` and `protocolVersionsImplementation` contract addresses.
+
+A sample TOML deploy config input for this step might look like:
+
+```toml
+[roles]
+proxyAdminOwner = "0x1234..."
+protocolVersionsOwner = "0x1234..."
+
+[superchain_config]
+guardian = "0x1234..."
+paused = false
+
+[protocol_versions]
+requiredVersion = "0.1.0"
+recommendedVersion = "0.1.0"
+```
+
+A sample TOML output for this step might look like:
+
+```toml
+superchainProxyAdmin = "0x1234..."
+superchainConfigProxy = "0x1234..."
+superchainConfigImpl = "0x1234..."
+protocolVersionsProxy = "0x1234..."
+protocolVersionsImpl = "0x1234..."
+
+# Everything above this comment are the outputs, below are the inputs.
+[roles]
+proxyAdminOwner = "0x1234..."
+protocolVersionsOwner = "0x1234..."
+
+[superchain_config]
+guardian = "0x1234..."
+paused = false
+
+[protocol_versions]
+requiredVersion = "0.1.0"
+recommendedVersion = "0.1.0"
+```
+
+## Step 2: Deploy Shared Implementation Contracts
+
+We assume the latest Fault Proofs release `op-contracts/v1.5.0`.
+The exact set of sub-steps and inputs may change with future releases.
+
+The sub-steps are deploying the following contracts:
+
+1. DisputeGameFactory
+2. AnchorStateRegistry
+3. DelayedWETH
+4. PreimageOracle
+5. MIPS
+6. OptimismPortal
+7. SystemConfig
+8. L1CrossDomainMessenger
+9. L1ERC721Bridge
+10. L1StandardBridge
+11. OptimismMintableERC20Factory
+
+The inputs for each sub-step are:
+
+1. None
+1. None (takes prior step's address as an input)
+1. `withdrawalDelaySeconds`
+1. `minProposalSizeBytes` `challengePeriodSeconds`
+1. None (takes prior step's address as an input)
+1. `_proofMaturityDelaySeconds` `disputeGameFinalityDelaySeconds`
+1. None
+1. None
+1. None
+1. None
+1. None
+
+The outputs for each sub-step are the contract addresses of each deploy.
+
+A sample TOML deploy config input for this step might look like:
+
+```toml
+[fault_proofs]
+withdrawalDelaySeconds = 1
+minProposalSizeBytes = 1
+challengePeriodSeconds = 1
+proofMaturityDelaySeconds = 1
+disputeGameFinalityDelaySeconds = 1
+```
+
+A sample TOML output for this step might look like:
+
+```toml
+disputeGameFactoryImpl = "0x123..."
+anchorStateRegistryImpl = "0x123..."
+delayedWETHImpl = "0x123..."
+preimageOracleImpl = "0x123..."
+mipsImpl = "0x123..."
+optimismPortalImpl = "0x123..."
+systemConfigImpl = "0x123..."
+l1CrossDomainMessengerImpl = "0x123..."
+l1ERC721BridgeImpl = "0x123..."
+l1StandardBridgeImpl = "0x123..."
+optimismMintableERC20FactoryImpl = "0x123..."
+
+# Everything above this comment are the outputs, below are the inputs.
+[fault_proofs]
+withdrawalDelaySeconds = 1
+minProposalSizeBytes = 1
+challengePeriodSeconds = 1
+proofMaturityDelaySeconds = 1
+disputeGameFinalityDelaySeconds = 1
+```
+
+## Step 3: Deploy OP Chain Contracts
+
+We assume the latest Fault Proofs release `op-contracts/v1.5.0`.
+The exact set of sub-steps and inputs may change with future releases.
+
+The sub-steps are:
+
```

Comment:
The Safe contract thing is missing I believe, there is some extra step in the `Deploy` contract for a named Safe, which we may need to integrate here. (Although I don't believe it's currently going into the op-e2e build of L1 chain state)

Reply:
This is an interesting point. The Safe architecture is technically peripheral to the protocol itself, and is a step you'd use to configure the ProxyAdmin owner role holder a certain way. I think we can leave it out of scope here for now, what do you think @maurelian?

Reply:
Agree, it should be safer and simpler to exclude the Safes that control the system. Any inputs to the system that originate from the Safes (either the ProxyAdmin owner or the Guardian) can be simulated by an EOA.